### ISSUE TITLE
Unit tests for Variants

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#690]
We need unit tests for variants, where we test the basic properties of objects defined in ga4gh/datamodel/variants.py outside of the context of input data. The tests created here should be similar to the new tests outlined in #683, #688, #687.

Open ``tests/unit/test_variants.py``, and add the following class:

```python
class TestAbstractVariantSet(unittest.TestCase):
    """
    Unit tests for the abstract variant set.
    """
    def setUp(self):
        self._backend = backend.AbstractBackend()
        self._dataset = datasets.AbstractDataset(self._backend)
        self._variantSet = variants.AbstractVariantSet(
            self._dataset, "testVariantSet")

    def testAddOneCallSet(self):
        self.assertEqual(self._variantSet.getNumCallSets(), 0)
        callSetName = "callSetName"
        self._variantSet.addCallSet(callSetName)
        self.assertEqual(self._variantSet.getNumCallSets(), 1)
        callSet = self._variantSet.getCallSetByIndex(0)
        self.assertEqual(
            self._variantSet.getCallSetByName(callSetName), callSet)
        self.assertEqual(                                         
            self._variantSet.getCallSet(callSet.getId()), callSet)   
        self.assertEqual(self._variantSet.getCallSets(), [callSet])
```

Run the tests by running ``nosetests -vs tests/unit/test_variants.py:TestAbstractVariantSet``

- Add another test ``testAddMultipleCallSets`` which is similar.

- Add tests to check that calling methods like getCallSet and getCallSetByName raise the correct exceptions (see an example in #683).

- Add a test for the hashVariant method.

**Comment by jsh2134** [comment]
Working on this!

**Comment by dcolligan** [comment]
Fixed in #739
